nrpy/helpers/safewrite.py

```diff
@@ -0,0 +1,87 @@
+"""
+The Safewrite class helps users see and understand changes to generated output files.
```

Comment:
Too vague.

Reply:
Also I really don't like the name of this python file or the class. Maybe ConditionalFileUpdater would be more appropriate?